Fix bower repo names, settings generation and prevent cache

kaizo opened the pull request.

vHanda committed:
Fix bower repo names

kaizo committed:
Fix typo in start.sh when generating enableUserRegistration variable from envars

This modifies the default value of the enableUserRegistration variable
in the platformSettings.js file with the value configured in the envars.

It was checking wrongly if the envar had content and overwriting the
value even if there where no content, generating an invalid format in
the js file and breaking the whole login page UX.

kaizo committed:
Prevent the javascript files being cached

We hash the file and append that to the filename, so we are sure that it
will change the filename when the content is changed.

jordi-eyeos committed:
Remove eyerun requestor references

Eyerun is not used anymore, so it's time to clean it!

jordins commented on grunt_tasks/hashAndReplaceFiles.js:
```diff
@@ -0,0 +1,51 @@
+var hashAndReplaceFiles = function (grunt, files, whereToReplace) {
```
Duplicated file from desktop, we should extract it into an npm library.

jordi-eyeos committed:
Remove some eyerun references

because it's not used anymore.

Commit:
Change id forgot password label.

I've changed the id with more appropiated name.

Commit:
Move error message outside the login form

Commit:
Remove attribute class because don't have any class name

Commit:
Move register button inside the form because it's more appropiated

Commit:
Add forgot password form

We've done this because we want to have the forgot password form
in the same page that login. We share a lot of styles and
param checkers

Commit:
Add reset password form

We've done this because we want to have the reset password form
in the same page that login. We share a lot of styles and
param checkers

Commit:
Change name shakeLogin to shakeBox

I've changed the name because with forgot password functionality
we want to reuse the function and it has to be more generic

Commit:
Change function with more specific name

We've decided to change the name preapreSubmit to
prepareLoginSubmit because we add two new submits: one for
forgotpassword to send mail and resetpassword.

Commit:
New error domain message function

This function is used for show an error when user not add the
domain

Commit:
New success message function

I've added this function to show the user that something
action in forgotPassword it's ok.

Commit:
Add functions for show/hide forgotPassword form

Commit:
Add functions for show/hide recoverPassword form

Commit:
Add functions for show/hide login form

Commit:
Add forgot password functionality

I've added this because we want that the user has the
possibility to change his password.

Commit:
Add reset password functionality

I've added this because we want that the user reset his
password.
This functionality only has beend accessible from a url
given in the user mail.

The url contains token and username.

Commit:
Added styles on success message

Commit:
Modified translator
- Added new method to message translation
- Added new tags to all localizations

Commit:
Remove recovery and change-pass

I've removed this files because i've separated the functionality
in two parts and i've reused the index.html.

I've created forgotPassword and ResetPassword. In the index.html
there are one form for every functionality. And the prepare.js
shows/hides the forms according the purpose.

Commit:
Add forgot object on login.test.js.

We've added this object for mocking it on login test.

Commit:
Added coverage for the function init on login.js

Commit:
Change forgotPassword info/error messages

Commit:
When user reset the password correctly redirect to main page

Commit:
Updated localization
- Using the object translator for localization messages
- Removing unused tags
- Fixing test
- Update Readme

Commit:
Removed elements to launch eyeos

Commit:
Applied localization into login

Commit:
Hide success message when error message is shown

Commit:
Parse response object for get the correct errors

Commit:
Applied localization into html of terms and coditions

Commit:
Removed tag degugger

Commit:
Remove email.html is not used

Commit:
Applied localization into forgot password

Commit:
Applied localization into recovery password

Commit:
Removed terms-and-conditions.html

Commit:
Added a new localization into forgot password

dodonpachi committed:
Update loc.login.it_IT.po

First IT localization

dodonpachi committed:
Update loc.login.es_ES.po

First es_ES localization

Commit:
Fixed translate ellipsis in localization

jordi-eyeos committed:
Fix applying platform settings (like language)

When platformSettings was hashed in build process it was not replaced in
the start.sh which applies different settings to login, like language.

This commit fixes it.

jordi-eyeos committed:
Fix loading correct translations

we don't have userInfo right now, so we need to get it from the platform
settings.

jordins commented on src/js/main.js:
```diff
@@ -46,36 +45,34 @@ require([
-			prepare.showLoading();
+
+		prepare.showLoading();
+		var params = prepare.getUrlParametersByNames(['username', 'token']);
```
please, get the url path (/password/reset)

absis replied:
Done

jordins commented on src/js/prepare.js:
```diff
@@ -241,20 +226,51 @@ define([
 		}
 	};
 
+	Prepare.prototype.getUrlParametersByNames = function(names) {
```
maybe use getURLParameter (redirector file)?

absis replied:
Done

jordins commented on src/js/translator.js:
```diff
@@ -50,32 +53,42 @@ define(["js/tr", "i18next"], function (tr) {
+	};
+
+	Translator.prototype.messageTranslation = function(message) {
+		return tr(message);
```
why this wrapper?

Commit:
Fixed Link style in forgot password

Commit:
Add browser detection and prevent access with not supported browsers

Commit:
Fixed generate settings file to run docker

Commit:
Add new param lang for forgotPassword call.

This param is used for send a forgot password email with the user language.

Commit:
Add some coverage

Commit:
Fixed apply styles into browserNotSupported

Commit:
Fixed apply styles into browserNotSupported

kaizo committed:
Add all HTML files to the list of files to modify when avoiding cache problems

We are adding a hash at the end of the platformSettings filename to
avoid cache problems. This file is being used in multiple places and
some where missing in the list of files to modify.

jordins committed:
Extract grunt build code to library

It's gonna be reused in desktop the same lib.

Commit:
Login refactor. Put styles and code in separate files

Commit:
Fixed mobile detection

Commit:
Fixed mobile detection

Commit:
Add new param on forgot request: domain.

I've added this new param because backend service needs it for
generate the forgotPassword url.

Commit:
Add some test coverage

Commit:
Check location.pathname is equal to /password/reset/.

I've added this check for redirect to recover password.

jordins commented on src/js/main.js:
```diff
@@ -80,10 +80,12 @@ require([
 
 		prepare.showLoading();
-		var params = prepare.getUrlParametersByNames(['username', 'token']);
+		if(location.pathname === "/password/reset/") {
```
Can this path be in settings?

absis replied:
Done

Commit:
Add pathname and requestParams to settings file

Commit:
Create new class called URL.

I've created this class because we've made a lot of operations with
url parts in many classes. So, I've moved this logical to a new class.

Commit:
User url.js instead of own url logic in redirector class.
I've removed this logic because I've created Url.js and this logic
is in this class.

Commit:
Move all reset password logic to recover.js.

I've decided to move the logic because the recover.js must be the
responsible.

Commit:
Remove getUrlParameters.

This logic is in the URL.js class.

Commit:
Add timeout for show success message when user recover his password

Commit:
Add new error message. Invalid domain.
